s4e-backend/src/main/java/pl/cyfronet/s4e/service/ProductCategoryService.java

```diff
@@ -0,0 +1,19 @@
+package pl.cyfronet.s4e.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import pl.cyfronet.s4e.properties.FileStorageProperties;
+
+@Service
+@RequiredArgsConstructor
+public class ProductCategoryService {
+    private final static String ICON_EXTENSION_WITH_DOT = ".svg";
+
+    private final FileStorageProperties fileStorageProperties;
+
+    public String getIconUrlBy(String iconName) {
+        return fileStorageProperties.getKeyPrefixProductsCategoriesIcons()
+                + iconName
+                + ICON_EXTENSION_WITH_DOT;
+    }
```

Comment:
This method should be named `getIconKey`, or something like that, as it's not an url or a path (see SavedViewService for an analogue). Also, `BasicProductCategoryResponse::getUrl` should be rather named `getIconKey` if that's what you want to return. In front-end you then just prepend `/static` and get a valid relative path.